core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,26 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Test that unknown field trigger ParsingException.
+     * To find the right position in the root query, we add a marker as `queryName` which
+     * all query builders support. The added bogus field after that should trigger the exception.
+     * Queries that allow arbitrary field names at this level need to override this test.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        assumeFalse("Skip test in unlikely event that we have _marker_ in some random string", testQuery.toString().contains("_marker_"));
+        testQuery.queryName("#marker#"); // to find root query to add additional bogus field there
+        String queryAsString = testQuery.toString().replace("\"#marker#\"", "\"#marker#\", \"bogusField\" : \"someValue\"");
```

Comment:
That's what the assumeFalse() is doing before we are setting the extra `queryName` field with the marker. You made me realize there is still a typo there, but other than that I think its safe. If there is no occurence of the maker before adding it via `queryName` then there is only one after.